### ISSUE TITLE
Access denied on AWS for atom-shell/node-v4.4.2-headers

### ISSUE THREAD

**Issue by SkyPol0** [Microsoft/vscode#1912]
Hi,

I'm trying to build from source.
Executing scripts\npm install command is outputting the current errors  : 
```
> native-keymap@0.1.2 install C:\Users\Paul Fasola\Documents\GitHub\vscode\node_modules\na
tive-keymap
> node-gyp rebuild


C:\Users\Paul Fasola\Documents\GitHub\vscode\node_modules\native-keymap>if not defined npm
_config_node_gyp (node "C:\Program Files (x86)\nodejs\node_modules\npm\bin\node-gyp-bin\\.
.\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node  rebuild )
gyp WARN install got an error, rolling back install
gyp ERR! configure error
gyp ERR! stack Error: 403 response downloading https://atom.io/download/atom-shell/v4.2.2/
node-v4.2.2-headers.tar.gz
gyp ERR! stack     at Request.<anonymous> (C:\Program Files (x86)\nodejs\node_modules\npm\
node_modules\node-gyp\lib\install.js:244:14)
gyp ERR! stack     at emitOne (events.js:82:20)
gyp ERR! stack     at Request.emit (events.js:169:7)
gyp ERR! stack     at Request.onRequestResponse (C:\Program Files (x86)\nodejs\node_module
s\npm\node_modules\request\request.js:998:10)
gyp ERR! stack     at emitOne (events.js:77:13)
gyp ERR! stack     at ClientRequest.emit (events.js:169:7)
gyp ERR! stack     at HTTPParser.parserOnIncomingClient (_http_client.js:415:21)
gyp ERR! stack     at HTTPParser.parserOnHeadersComplete (_http_common.js:88:23)
gyp ERR! stack     at Socket.socketOnData (_http_client.js:305:20)
gyp ERR! stack     at emitOne (events.js:77:13)
gyp ERR! System Windows_NT 10.0.10240
gyp ERR! command "C:\\Program Files (x86)\\nodejs\\node.exe" "C:\\Program Files (x86)\\nod
ejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\Paul Fasola\Documents\GitHub\vscode\node_modules\native-keymap
gyp ERR! node -v v4.2.2
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok

> windows-mutex@0.2.0 install C:\Users\Paul Fasola\Documents\GitHub\vscode\node_modules\wi
ndows-mutex
> node-gyp rebuild


C:\Users\Paul Fasola\Documents\GitHub\vscode\node_modules\windows-mutex>if not defined npm
_config_node_gyp (node "C:\Program Files (x86)\nodejs\node_modules\npm\bin\node-gyp-bin\\.
.\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node  rebuild )
gyp WARN install got an error, rolling back install
gyp ERR! configure error
gyp ERR! stack Error: 403 response downloading https://atom.io/download/atom-shell/v4.2.2/
node-v4.2.2-headers.tar.gz
gyp ERR! stack     at Request.<anonymous> (C:\Program Files (x86)\nodejs\node_modules\npm\
node_modules\node-gyp\lib\install.js:244:14)
gyp ERR! stack     at emitOne (events.js:82:20)
gyp ERR! stack     at Request.emit (events.js:169:7)
gyp ERR! stack     at Request.onRequestResponse (C:\Program Files (x86)\nodejs\node_module
s\npm\node_modules\request\request.js:998:10)
gyp ERR! stack     at emitOne (events.js:77:13)
gyp ERR! stack     at ClientRequest.emit (events.js:169:7)
gyp ERR! stack     at HTTPParser.parserOnIncomingClient (_http_client.js:415:21)
gyp ERR! stack     at HTTPParser.parserOnHeadersComplete (_http_common.js:88:23)
gyp ERR! stack     at Socket.socketOnData (_http_client.js:305:20)
gyp ERR! stack     at emitOne (events.js:77:13)
gyp ERR! System Windows_NT 10.0.10240
gyp ERR! command "C:\\Program Files (x86)\\nodejs\\node.exe" "C:\\Program Files (x86)\\nod
ejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\Paul Fasola\Documents\GitHub\vscode\node_modules\windows-mutex
gyp ERR! node -v v4.2.2
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok

> alexandrudima-oniguruma@6.0.1 install C:\Users\Paul Fasola\Documents\GitHub\vscode\node_
modules\vscode-textmate\node_modules\alexandrudima-oniguruma
> node-gyp rebuild


C:\Users\Paul Fasola\Documents\GitHub\vscode\node_modules\vscode-textmate\node_modules\ale
xandrudima-oniguruma>if not defined npm_config_node_gyp (node "C:\Program Files (x86)\node
js\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild
 )  else (node  rebuild )
gyp WARN install got an error, rolling back install
gyp ERR! configure error
gyp ERR! stack Error: 403 response downloading https://atom.io/download/atom-shell/v4.2.2/
node-v4.2.2-headers.tar.gz
gyp ERR! stack     at Request.<anonymous> (C:\Program Files (x86)\nodejs\node_modules\npm\
node_modules\node-gyp\lib\install.js:244:14)
gyp ERR! stack     at emitOne (events.js:82:20)
gyp ERR! stack     at Request.emit (events.js:169:7)
gyp ERR! stack     at Request.onRequestResponse (C:\Program Files (x86)\nodejs\node_module
s\npm\node_modules\request\request.js:998:10)
gyp ERR! stack     at emitOne (events.js:77:13)
gyp ERR! stack     at ClientRequest.emit (events.js:169:7)
gyp ERR! stack     at HTTPParser.parserOnIncomingClient (_http_client.js:415:21)
gyp ERR! stack     at HTTPParser.parserOnHeadersComplete (_http_common.js:88:23)
gyp ERR! stack     at Socket.socketOnData (_http_client.js:305:20)
gyp ERR! stack     at emitOne (events.js:77:13)
gyp ERR! System Windows_NT 10.0.10240
gyp ERR! command "C:\\Program Files (x86)\\nodejs\\node.exe" "C:\\Program Files (x86)\\nod
ejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\Paul Fasola\Documents\GitHub\vscode\node_modules\vscode-textmate\nod
e_modules\alexandrudima-oniguruma
gyp ERR! node -v v4.2.2
gyp ERR! node-gyp -v v3.0.3
gyp ERR! not ok
npm WARN optional dep failed, continuing windows-mutex@0.2.0
npm ERR! Windows_NT 10.0.10240
npm ERR! argv "C:\\Program Files (x86)\\nodejs\\node.exe" "C:\\Program Files (x86)\\nodejs
\\node_modules\\npm\\bin\\npm-cli.js" "install"
npm ERR! node v4.2.2
npm ERR! npm  v2.14.7
npm ERR! code ELIFECYCLE

npm ERR! native-keymap@0.1.2 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the native-keymap@0.1.2 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the native-keymap package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls native-keymap 
```

Here's the npm-debug file : http://pastebin.com/8N3SxfX6

Oh, and I've already checked  my firewall ;)

**Comment by joaomoreno** [comment]
Wanna give it another try? Might be a problem in the `atom.io` server.

**Comment by SkyPol0** [comment]
Still denied :( 

I'm getting this error since 3 days

**Comment by joaomoreno** [comment]
Something is wrong in your setup, since in my machine the url `node-gyp` requests is `https://atom.io/download/atom-shell/v0.35.6/node-v0.35.6.tar.gz`.

Yours seems to be `https://atom.io/download/atom-shell/v4.2.2/
node-v4.2.2-headers.tar.gz`.

This means that the [automatic Electron version detection](https://github.com/Microsoft/vscode/blob/master/scripts/npm.bat#L5) isn't working in your system. Do you have PowerShell installed?

**Comment by SkyPol0** [comment]
Yes I've got it and I've tried to execute the scripts\npm install command (instead of doing this with the command prompt), still the same error!

**Comment by joaomoreno** [comment]
In the Command Prompt, what do you get when running the following snippet from the `vscode` directory?

```powershell -Command "(Get-Content -Raw package.json | ConvertFrom-Json).electronVersion"```

**Comment by SkyPol0** [comment]
-> 0.35.5

**Comment by joaomoreno** [comment]
Can you delete both these folders and try again?

- `vscode\node_modules`
- `.node-gyp` located in your user's home folder

**Comment by SkyPol0** [comment]
I have tried again after those deletions, still failing (same errors)  :unamused:

**Comment by joaomoreno** [comment]
Replace the contents of `scripts\npm.bat` with [this](https://gist.github.com/joaomoreno/bfc1b4efb7f1b069df78) and let me know what you get when you run it.

**Comment by SkyPol0** [comment]
"https://atom.io/download/atom-shell"
ECHO command disabled.
"ia32"

**Comment by joaomoreno** [comment]
I've updated the gist, can you try [that one](https://gist.github.com/joaomoreno/bfc1b4efb7f1b069df78)?

**Comment by SkyPol0** [comment]
Oh sorry, I've forgotten to notify yourself that I already noticed about the @echo off and tested. 
Still the same output unfortunately.

**Comment by joaomoreno** [comment]
I'm sorry, I'm really not sure why [line 5](https://github.com/Microsoft/vscode/blob/master/scripts/npm.bat#L5) doesn't seem to work for you. If you replace it with the following code, it will work:

```
set npm_config_target="0.35.6"
```

That version needs to match the same as in our [`package.json`](https://github.com/Microsoft/vscode/blob/master/package.json#L4).

I'll close this issue since it is a Command Prompt issue and you seem to be the only person that actually hit it. If you eventually figure out the real reason why it doesn't work, please let us know!

**Comment by joaomoreno** [comment]
Hey @SkyPol0, this might've been fixed with #2118. Pull and give that a try!

**Comment by dbaeumer** [comment]
Setting the verified label without actually verifying since it is not reproducible.
